### DISCUSSION TITLE
Wildcard CNAME causes unrelated content to display

### DISCUSSION THREAD

**Discussion by jspiewak** [community/community#22903] [Other Feature Feedback, Questions, & Ideas]
<p>We have a wildcard CNAME pointed to our GitHub Pages, e.g. *.example.com points to <a href="http://example.github.io" rel="nofollow noopener">example.github.io</a>.</p>
<p>When navigation to a certain subdomains, e.g. ftp and email, for which there are no pages, instead of the expected 404, we see content from “PDF Books Catalog”.</p>
<p>Searching Google, this is apparently not isolated to just us: <a href="http://lmgtfy.com/?q=PDF+Books+Catalog" rel="nofollow noopener">http://lmgtfy.com/?q=PDF+Books+Catalog</a></p>
<p>Today we received a DMCA take down notice for this.</p>
<p>Is anyone else actively experiencing this?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for writing in about this, and sorry for the trouble this has caused!</p>
<p>Setting up a wildcard DNS record will allow anyone to host a GitHub Pages site at one of your subdomains. This is <a href="https://help.github.com/articles/setting-up-a-custom-subdomain/" rel="nofollow noopener">something we specifically warn against in our documentation</a>. As we don’t have any way of verifying domain ownership over DNS the only security method we can recommend is explicitly specifying the subdomain records you need and pointing them at GitHub.</p>
<p>To fix this you’ll need to remove your wildcard record and create an explicit <code>CNAME</code> record for each subdomain you want to use.</p>
<p>Additionally, it would also be very helpful if you could send a list of any affected subdomains you’ve noticed in to private GitHub Support at <a href="https://github.com/contact" rel="nofollow noopener">https://github.com/contact</a>.</p>
<p>Thanks!</p>
